Server-side TLS

tcsc opened the pull request:
Adds server-side TLS to the `levityd` daemon, including
 * changing the server to require TLS connections by default
 * Adds CA override to client in order to use self-signed certs with
   the server TLS
 * certificate generation tasks in makefile
 * system tests that exercise TLS error cases

Note that this patch checks the certificates into the repository, along
with the code. This is the quickest path to getting CI working for this
project, but not necessarily the way I would do things on a production
system.

You can re-generate all of the CA and server cerificates with `make
certs`

tcsc committed:
Server-side TLS

Adds server-side TLS to the `levityd` daemon, including
 * changing the server to require TLS connections by default
 * Adds CA override to client in order to use self-signed certs with
   the server TLS
 * certificate generation tasks in makefile
 * system tests that exercise TLS error cases

Note that this patch checks the certificates into the repository, along
with the code. This is the quickest path to getting CI working for this
project, but not necessarily the way I would do things on a production
system.

You can re-generate all of the CA and server cerificates with `make
certs`

fspmarshall commented on cmd/system_test.go:
```diff
@@ -35,8 +35,12 @@ func (m *outputMonitor) Write(p []byte) (int, error) {
 func startDaemon(port int) (*daemon, error) {
-	addr := fmt.Sprintf("127.0.0.1:%d", port)
-	cmd := exec.Command("levityd", addr)
+	addr := fmt.Sprintf("0.0.0.0:%d", port)
```
Why bind `0.0.0.0`?  Seems dangerous.

russjones replied:
Interesting approach to integration testing, I would have just created and started the server process, but this tests process startup like the user would start it which is nice.

Agree with @fspmarshall though, tests should run on loopback. Also might want to do `127.0.0.1:0` to let the OS select a high random port.

tcsc replied:
Done

tcsc committed:
Address review concerns

Updates the integration tests to run the `levityd` server on the
loopback address, rather than the remotely-accessable 0.0.0.0.

Also lets the OS pick the port, rather than a hardcoded value, which
should let tests run concurrently. This requires some surgery to the
test harness in order to scan the server's log outputand extract the
assigned port.
